File: ExpenseTracker_backend/src/main/java/com/expensetracker/service/income/IncomeService.java
package com.expensetracker.service.income;

import java.util.List;

import com.expensetracker.dto.IncomeDTO;
import com.expensetracker.entity.Income;


public interface IncomeService {
	
	Income postIncome(IncomeDTO incomeDTO);	
	
	List<IncomeDTO> getAllIncomes();
	
	Income updateIncome(Long id, IncomeDTO incomeDTO);
	
	IncomeDTO getIncomeById(Long id);
	
	void deleteIncome(Long id);
	
	

}

File: ExpenseTracker_frontend/expensetrackerwebapp/src/app/services/income/income.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';

// const baseUrl = 'http://localhost:5001/';
const baseUrl = 'http://expensetracker-env.eba-vp6hmw64.eu-north-1.elasticbeanstalk.com/';

@Injectable({
  providedIn: 'root'
})
export class IncomeService {

  constructor(private http:HttpClient) { }

  postIncome(incomeDTO:any){
    return this.http.post(baseUrl+"api/income",incomeDTO);
  }

  getAllIncomes(){
    return this.http.get(baseUrl+"api/income/all");
  }

  getIncomeById(id:number){
    return this.http.get(baseUrl+"api/income/"+id);
  }

  updateIncome(id:number,incomeDTO:any){
    return this.http.put(baseUrl+"api/income/"+id,incomeDTO);
  }
  
  deleteIncome(id:number){
    return this.http.delete(baseUrl+"api/income/"+id);
  }
  
}
